OpenRA.Mods.Common/UtilityCommands/Utilities.cs

```diff
@@ -7,7 +7,10 @@ namespace OpenRA.Mods.Common.UtilityCommands
 {
 	public static class Utilities
 	{
-		public static MiniYamlNode GetTopLevelNodeByKey(ModData modData, string actorType, string mapPath = null)
+		public static MiniYamlNode GetTopLevelNodeByKey(ModData modData, string actorType,
+			Func<Manifest, string[]> manifestPropertySelector,
+			Func<Map, MiniYaml> mapPropertySelector = null,
```

Comment:
Defaulting these parameters buys you nothing. You provide them at all call sites (the one here, and the new one you introduce in the next commit)